### PR TITLE
[1LP][RFR] Cloud Intel / Reports widgetastic conversion. Part 3

### PR DESCRIPTION
**Pull request by quarckster**
Purpose
=================

This PR is a part of set of Cloud Intel / Reports widgetastic conversion PRs. Cloud Intel / Report / Dashboards have been converted to widgetastic.

{{pytest: -v -k "test_widgets_on_dashboard"}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/intelligence/reports/dashboards.py [hunk 1]
```diff
@@ -37,111 +168,154 @@ def group(self):
...
 
     def update(self, updates):
-        navigate_to(self, 'Edit')
-        fill(self.form, updates, action=form_buttons.save)
-        flash.assert_no_errors()
+        """Update this Dashboard in the UI.
+
+        Args:
+            updates: Provided by update() context manager.
+            cancel: Whether to cancel the update (default False).
```
Comment (izapolsk) [inline cfme/intelligence/reports/dashboards.py]: there is no cancel param

File: cfme/intelligence/reports/dashboards.py [hunk 2]
```diff
@@ -37,111 +168,154 @@ def group(self):
...
-    ])
-    reset_button = "//*[@title='Reset Dashboard Widgets to the defaults']"
-    pretty_attrs = ['title', 'widgets']
+
+    pretty_attrs = ["name", "title", "widgets"]
 
     def __init__(self, title=None, locked=None, widgets=None, appliance=None):
         Navigatable.__init__(self, appliance)
-        self.title = title
+        self.title = title or "Default Dashboard"
```
Comment (izapolsk) [inline cfme/intelligence/reports/dashboards.py]: please move this to __init__

File: cfme/intelligence/reports/dashboards.py [hunk 3]
```diff
@@ -37,111 +168,154 @@ def group(self):
...
+    def update(self, updates):
+        """Update Default Dashboard in the UI.
 
-@navigator.register(DefaultDashboard, 'All')
-@navigator.register(Dashboard, 'All')
-class DashboardAll(CFMENavigateStep):
-    prerequisite = NavigateToObject(Report, 'Dashboards')
+        Args:
+            updates: Provided by update() context manager.
+            cancel: Whether to cancel the update (default False).
```
Comment (izapolsk) [inline cfme/intelligence/reports/dashboards.py]: ^^

File: cfme/dashboard.py [hunk 4]
```diff
@@ -53,8 +53,10 @@ def reset_widgets(cancel=False):
     Args:
         cancel: Set whether to accept the popup confirmation box. Defaults to ``False``.
     """
+    navigate_to(Server, 'Dashboard')
     sel.click(page.reset_widgets_button, wait_ajax=False)
```
Comment (quarckster) [inline cfme/dashboard.py]: AFAIK @mfalesni is doing it in #5052

File: cfme/tests/intelligence/reports/test_widgets.py [hunk 5]
```diff
@@ -76,7 +76,7 @@ def _finalize():
         with update(dashboard):
             dashboard.widgets = default_widgets
     request.addfinalizer(_finalize)
-    dashboard.reset_widgets()
+    reset_widgets()
```
Comment (psav) [inline cfme/tests/intelligence/reports/test_widgets.py]: I have an issue with this function, it is not tied to an appliance object at all, what can we do about this?

Comment (quarckster) [inline cfme/tests/intelligence/reports/test_widgets.py]: @psav I can propose only some temporary widgetastic view only for this button, which will be replaced during cfme/dashboard.py conversion.

**Pull request merged**